Content.Server/Construction/PartExchangerSystem.cs

```diff
@@ -83,9 +142,8 @@ private void OnDoAfter(EntityUid uid, PartExchangerComponent component, DoAfterE
             storage.Storage.Insert(unused.Owner);
             _storage.Insert(uid, unused.Owner, null, false);
         }
-        _construction.RefreshParts(args.Args.Target.Value, machine);
 
-        args.Handled = true;
```

Comment:
pretty sure you still want this at the end

Reply:
It's at the end of the OnDoAfter method at line 54.
```
        TryExchangeMachineParts(args.Args.Target.Value, storage, machineParts);
        TryConstructMachineParts(args.Args.Target.Value, storage, machineParts);

        args.Handled = true;
    }
```